tests/neuro_san/client/data_driven_agent_test.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import Any
+from typing import Dict
+from typing import Generator
+from typing import List
+from typing import Union
+
+from datetime import datetime
+
+from leaf_common.persistence.easy.easy_hocon_persistence import EasyHoconPersistence
+
+from neuro_san.client.agent_session_factory import AgentSessionFactory
+from neuro_san.client.streaming_input_processor import StreamingInputProcessor
+from neuro_san.interfaces.agent_session import AgentSession
+from neuro_san.internals.utils.file_of_class import FileOfClass
+from neuro_san.message_processing.basic_message_processor import BasicMessageProcessor
+from neuro_san.session.direct_agent_session import DirectAgentSession
+
+from tests.neuro_san.client.agent_evaluator import AgentEvaluator
+from tests.neuro_san.client.agent_evaluator_factory import AgentEvaluatorFactory
+from tests.neuro_san.client.assert_forwarder import AssertForwarder
+
+
+class DataDrivenAgentTest:
+    """
+    Abstract test class whose subclasses define hocon file to parse as test cases.
+    """
+
+    TEST_KEYS: List[str] = ["text", "sly_data"]
+
+    def __init__(self, asserts: AssertForwarder, fixtures: FileOfClass = None):
+        """
+        Constructor
+        :param asserts: The AssertForwarder instance to use to integrate failures
+                        back into the test system.
+        :param fixtures: Optional path to the fixtures root.
+        """
+        self.asserts: AssertForwarder = asserts
+        self.fixtures: FileOfClass = fixtures
+        if self.fixtures is None:
+            self.fixtures = FileOfClass(__file__, path_to_basis="../../fixtures")
+
+    def one_test(self, hocon_file: str):
```

Comment:
Engine that drives the reading of a single hocon file for a round of testing a single agent.